src/components/10-atoms/heading/index.js

```diff
@@ -30,16 +30,31 @@ class AXAHeading extends LitElement {
     // this functions applies default values per type and verifies if
     // the HTML attribute has been set before defining the custom element
     applyDefaults(this);
+
+    this.topBottomMarginsByRank = {
+      1: '20px',
+      2: '18px',
+      3: '16px',
+      4: '14px',
+      5: '12px',
+      6: '10px',
+    };
   }
 
   render() {
     const secondaryVariant =
       this.variant === 'secondary' ? 'a-heading--secondary' : '';
 
     const template = `
-      <h${this.rank} class="a-heading ${secondaryVariant}">
-        <slot></slot>
-      </h${this.rank}>
+    <style>
+      :host {
+        display: block;
+        margin: ${this.topBottomMarginsByRank[this.rank]} 0;
```

Comment:
I think generated style tags with variable content are hacky, even though I get the idea and it's hidden under ShadowDOM.

Since this is per-instance variation, I recommend `this.style.marginTop = TOP_BOTTOM_MARGINS_BY_RANK[this.rank];` and similar for `marginTop`.

Reply:
If my suggestion is followed, the remaining `:host { display: content; }` should be moved out to index.scss.

Reply:
Genius! This is way better.